src/lavinmq/etcd.cr

```diff
@@ -377,24 +379,22 @@ module LavinMQ
     end
 
     private def raise_if_error(json)
-      if error = json["error"]?
-        Log.debug { "etcd error: #{error}" }
-        error_msg =
-          if errorh = error.as_h?
-            errorh["message"].as_s
-          else
-            error.as_s
-          end
-        case error_msg
-        when "error reading from server: EOF"
-          raise IO::EOFError.new(error_msg)
-        when "etcdserver: no leader"
-          raise NoLeader.new(error_msg)
-        when "etcdserver: lease already exists"
-          raise LeaseAlreadyExists.new
-        else
-          raise Error.new error_msg
-        end
+      error_msg = json.dig?("error", "message").try(&.as_s) ||
+                  json["error"]?.try(&.as_s?) ||
+                  json["message"]?.try(&.as_s)
```

Comment:
Any risk of a response with `{"message": "success"}`  or similiar, that will in this case generate an exception even though it was success.

Reply:
Yes, let's see how we can tighten this.

Reply:
https://etcd.io/docs/v3.6/dev-guide/api_concurrency_reference_v3/#service-election-serveretcdserverapiv3electionv3electionpbv3electionproto

Cannot find the specific response.